membership-attribute-service/app/actions/CommonActions.scala

```diff
@@ -22,24 +22,30 @@ class CommonActions(touchpointBackends: TouchpointBackends, bodyParser: BodyPars
   def noCache(result: Result): Result = NoCache(result)
 
   val NoCacheAction = resultModifier(noCache)
-  def AuthAndBackendViaAuthLibAction(requiredScopes: List[AccessScope]) =
+  def AuthAndBackendViaAuthLibAction(requiredScopes: List[AccessScope]): ActionBuilder[AuthenticatedUserAndBackendRequest, AnyContent] =
     NoCacheAction andThen new AuthAndBackendViaAuthLibAction(touchpointBackends, requiredScopes, isTestUser)
-  def AuthAndBackendViaIdapiAction(howToHandleRecencyOfSignedIn: HowToHandleRecencyOfSignedIn) =
+  def AuthAndBackendViaIdapiAction(howToHandleRecencyOfSignedIn: HowToHandleRecencyOfSignedIn): ActionBuilder[AuthAndBackendRequest, AnyContent] =
     NoCacheAction andThen new AuthAndBackendViaIdapiAction(touchpointBackends, howToHandleRecencyOfSignedIn, isTestUser)
 
+  def AuthAndBackendViaBothAction(
+      howToHandleRecencyOfSignedIn: HowToHandleRecencyOfSignedIn,
+      requiredScopes: List[AccessScope],
+  ): ActionBuilder[AuthenticatedUserAndBackendRequest, AnyContent] =
+    NoCacheAction andThen
+      new AuthAndBackendViaIdapiAction(touchpointBackends, howToHandleRecencyOfSignedIn, isTestUser) andThen
+      new AuthAndBackendViaAuthLibAction(touchpointBackends, requiredScopes, isTestUser)
+
   private def resultModifier(f: Result => Result) = new ActionBuilder[Request, AnyContent] {
     override val parser = bodyParser
     override val executionContext = ex
     def invokeBlock[A](request: Request[A], block: (Request[A]) => Future[Result]) = block(request).map(f)
   }
 }
 
-class BackendRequest[A](val touchpoint: TouchpointComponents, request: Request[A]) extends WrappedRequest[A](request)
-
 class AuthenticatedUserAndBackendRequest[A](
-    val user: Either[AuthenticationFailure, UserFromToken],
+    val user: UserFromToken,
```

Comment:
You seem to be losing some information here about the authentication result

Reply:
If you look at where an instance of this class is created, there's no code path that returns Left.

Reply:
Instead the whole response is turned into Unauthorized or such.

Reply:
Here:
https://github.com/guardian/members-data-api/blob/main/membership-attribute-service/app/actions/AuthAndBackendViaAuthLibAction.scala#L20-L32